File: leetcode/Python/maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        sol, um = 100000, defaultdict(int)
        for c in text:
            um[c]+=1
        for c in 'balon':
            if c=='l' or c=='o':
                sol=min(sol,um[c]//2)
            else:
                sol=min(sol,um[c])
        return sol

File: leetcode/C++/maximum-number-of-balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        int sol=10000;
        string s = "balon";
        unordered_map<char,int> um;
        for(char & c : text) ++um[c];
        for(char & c : s) {
            if(c=='l' || c=='o') sol=min(sol,um[c]/2);
            else sol=min(sol,um[c]);
        }
        return sol;
    }
};